ls-alg/main/Main.java

```diff
@@ -0,0 +1,162 @@
+package main;
```

Comment:
Let's rename this file `AvodataSolverTool.java` since it's a runnable (hence tool) and `AvodataSolver` seems like a good name, unless you can think of a better one.